tsconfig.json

```diff
@@ -11,6 +11,7 @@
     "exactOptionalPropertyTypes": true,
     "baseUrl": ".",
     "skipLibCheck": true,
+    "noEmit": true,
```

Comment:
Do we need this even if we have `--noEmit` option?

Reply:
not anymore

Reply:
sorry i misunderstood, yes we need it because this flag is only allowed when --noEmit or --emitDeclarationOnly is enabled, since these import paths would not be resolvable at runtime in JavaScript output files.

https://www.typescriptlang.org/tsconfig#allowImportingTsExtensions

Reply:
We used to have `--noEmit` in package.json and no "noEmit" in tsconfig.json. Does it cause any errors with this PR's change?
I mean I'm aware of https://www.typescriptlang.org/tsconfig#allowImportingTsExtensions and it's working before this PR, so I'm confused.

Reply:
it'ok, we are bundling with rollup instead of tsc. 
if you remove the `--noEmit` option in the pretest script, you will be able to see the complaint in the console.

Reply:
VSCode(tsserver) will show an error in the tsconfig.json

Reply:
![image](https://github.com/pmndrs/jotai/assets/13805071/a2c03e87-14c6-47d6-93c5-e85b83684ac7)

Reply:
Alright! thanks for the clarification!